### ISSUE TITLE
Doesn't recognise HOSTNAME set in iredmail-docker.conf

### ISSUE THREAD

**Issue by Z0lid** [iredmail/dockerized#68]
Hi, when I try to send an email it gives me this error:

> The following recipient(s) cannot be reached:
>       'other-person@gmail.com' on _thisdate_
>             Server error: '451 4.3.0 <me@myhostname.com>: Temporary lookup failure'

When I try to run any command using `docker run -it iredmail/mariadb:stable sh -c`, it gives me this error:

> [iRedMail] ERROR: HOSTNAME is not a fully qualified domain name (FQDN).
> [iRedMail] ERROR: Please fix it in 'iredmail-docker.conf' first.

My hostname is set (and is correct) in the `iredmail-docker.conf` file and in the `docker-compose.yaml`
Any help would be appreciated

**Comment by iredmail** [comment]
You should run it with `docker run --env-file iredmail-docker.conf ...`.

**Comment by Z0lid** [comment]
Sorry, I meant to run `docker exec -it` instead of `docker run`, since I already have the container working from a `docker-compose.yaml`, and I can receive emails, the problem is when sending them, this error is "sent" to me in my mailbox, and the email doesnt send:
> The following recipient(s) cannot be reached:
> 'other-person@gmail.com' on _thisdate_
> Server error: '451 4.3.0 me@myhostname.com: Temporary lookup failure'

**Comment by iredmail** [comment]
- When you run it on command line, please load the `iredmail-docker.conf` with the `--env-file` argument.
- Make sure docker-compose.yml loads the `iredmail-docker.conf` file.

Please do some checks to make sure they're loaded.

**Comment by Z0lid** [comment]
I'm sure the `iredmail-docker.conf` is loaded because when I run `docker exec -it iredmail sh -c 'printenv FIRST_MAIL_DOMAIN'`, it prints me the FIRST_MAIL_DOMAIN env var that I set in `iredmail-docker.conf`

**Comment by Z0lid** [comment]
I have figured out that my outbound port 25 is blocked, where can I edit the postfix configuration to set up the SMTP relay?

**Comment by iredmail** [comment]
Add/Update bash shell script `/opt/iredmail/custom/postfix/custom.sh`, run `postconf -e <var>=<value>` in this script to update `/etc/postfix/main.cf`.

**Comment by Z0lid** [comment]
> Jul 20 16:05:01 mail postfix/proxymap[877]: warning: connect to mysql server 127.0.0.1:3306: Access denied for user 'vmail'@'localhost' (using password: YES)
> Jul 20 16:05:01 mail postfix/trivial-rewrite[885]: warning: virtual_mailbox_domains: proxy:mysql:/etc/postfix/mysql/virtual_mailbox_domains.cf: table lookup problem

I'm getting these table lookup problems in `/var/log/mail.log`, it seems like postfix cant connect to the database for some reason, I haven't touched any of the database settings, how can I solve this?

**Comment by iredmail** [comment]
Check the SQL user privilege: `vmail@localhost`, and its password (you can find it in `/etc/postfix/mysql/*.cf`).

**Comment by w1ld3r** [comment]
Having the same issue, I've looked in `/etc/postfix/mysql/virtual_mailbox_domains.cf` for `vmail`'s password and tried to authenticate localy, unfortunately it didn't work:
```bash
mysql -u vmail -p
Enter password: 
ERROR 1045 (28000): Access denied for user 'vmail'@'localhost' (using password: YES)
```
Using the mysql root account I've been able to login to the mariadb server (`mysql -u root -p`) to update the vmail account password with:
```sql
UPDATE mysql.user SET Password=PASSWORD('Password_From_virtual_mailbox_domains.cf') WHERE User='vmail';
FLUSH PRIVILEGES;
```
After the change, I was able to authenticate localy to the `vmail`'s sql account using its password but not after a restart of the container.

It seems the `./scripts/mariadb/pre_start/0-vmail_db.sh` script at each startup creates the vmail user using the `VMAIL_DB_PASSWORD` variable environment has password. Defining it in the env file `iredmail-docker.conf` and restarting the container didn't changes the `vmail` password in `virtual_mailbox_domains.cf` nore allowing to authenticate using the two passwords known.

So the issue must be somewhere else ...
